src/tab-controller/tests/functional/TabController.ts

```diff
@@ -7,6 +7,7 @@ import * as css from '../../../theme/tab-controller.m.css';
 import { uuid } from '@dojo/framework/core/util';
 
 const axe = services.axe;
+const DELAY = 1000;
```

Comment:
It anyone knows of an async way of waiting for `click` and `type` to finish before continuing please let me know. I would rather not use the synchronous `sleep` method.

Reply:
The general idea is to identify something that changes as a result of `click` or `type` and then use [`find`](https://theintern.io/docs.html#Leadfoot/2/api/Session/find), [`executeAsync`](https://theintern.io/docs.html#Leadfoot/2/api/Session/executeasync), or [`pollUntil`](https://theintern.io/docs.html#Leadfoot/2/api/helpers%2FpollUntil) to poll for the condition.

I have some work in [my branch](https://github.com/dojo/widgets/compare/master...msssk:test-async-improvements), but the 'Validated TextInput should update style based on validity' test is still failing. It does successfully demonstrate using `pollUntilTruthy` which replaces a fixed 250ms sleep with a poll that will finish as soon as the DOM updates (or potentially wait as long as 5s for a severely resource-constrained test environment to respond to the action).

Reply:
Hmm ok I see what your doing here, let me check it out for my fixes here